modules/fee-market/src/s2s/payment.rs

```diff
@@ -216,12 +244,56 @@ where
 	rewards_book
 }
 
-/// Do slash for absent assigned relayers
-pub(crate) fn do_slash<T: Config<I>, I: 'static>(
+/// Calculate the reward for the order which has been confirmed in time.
+pub(crate) fn cal_rewards_before_deadline<T: Config<I>, I: 'static>(
+	slot_relayer_id: &T::AccountId,
+	message_relayer_id: &T::AccountId,
+	confirm_relayer_id: &T::AccountId,
+	total_reward: BalanceOf<T, I>,
+	base_fee: BalanceOf<T, I>,
+	reward_item: &mut RewardItem<T::AccountId, BalanceOf<T, I>>,
+) {
+	// message fee - base fee => treasury_sum
+	reward_item.to_treasury = Some(total_reward.saturating_sub(base_fee));
+
+	// AssignedRelayersRewardRatio * base fee => slot relayer
+	let slot_relayer_reward = T::AssignedRelayersRewardRatio::get() * base_fee;
+	reward_item.to_slot_relayer = Some((slot_relayer_id.clone(), slot_relayer_reward));
+
+	let bridger_relayers_reward = base_fee.saturating_sub(slot_relayer_reward);
```

Comment:
```suggestion
	let message_and_confirm_reward = base_fee.saturating_sub(slot_relayer_reward);
```

Deserve better naming